pallets/issue/src/tests.rs

```diff
@@ -624,3 +624,338 @@ fn test_request_issue_reset_interval_and_succeeds_with_rate_limit() {
 		assert_eq!(<crate::CurrentVolumeAmount<Test>>::get(), BalanceOf::<Test>::zero());
 	})
 }
+mod integration_tests {
+	use super::*;
+	use orml_traits::MultiCurrency;
+	use pooled_rewards::RewardsApi;
+
+	fn get_reward_for_vault(vault: &DefaultVaultId<Test>, reward_currency: CurrencyId) -> Balance {
+		<<Test as fee::Config>::VaultRewards as RewardsApi<
+			CurrencyId,
+			DefaultVaultId<Test>,
+			Balance,
+			CurrencyId,
+		>>::compute_reward(&vault.collateral_currency(), vault, reward_currency)
+		.unwrap()
+	}
+
+	fn get_balance(currency_id: CurrencyId, account: &AccountId) -> Balance {
+		<orml_currencies::Pallet<Test> as MultiCurrency<AccountId>>::free_balance(
+			currency_id,
+			account,
+		)
+	}
+	fn nominate_vault(nominator: AccountId, vault: DefaultVaultId<Test>, amount: Balance) {
+		let origin = RuntimeOrigin::signed(vault.account_id);
+
+		assert_ok!(<nomination::Pallet<Test>>::opt_in_to_nomination(
+			origin.clone(),
+			vault.clone().currencies
+		));
+
+		let nominator_origin = RuntimeOrigin::signed(nominator);
+		assert_ok!(<nomination::Pallet<Test>>::deposit_collateral(nominator_origin, vault, amount));
+	}
+
+	fn register_vault_with_collateral(vault: &DefaultVaultId<Test>, collateral_amount: Balance) {
+		let origin = RuntimeOrigin::signed(vault.account_id);
+
+		assert_ok!(VaultRegistry::register_public_key(origin.clone(), DEFAULT_STELLAR_PUBLIC_KEY));
+		assert_ok!(VaultRegistry::register_vault(
+			origin,
+			vault.currencies.clone(),
+			collateral_amount
+		));
+	}
+	fn wrapped_with_custom_curr(amount: u128, currency: CurrencyId) -> Amount<Test> {
+		Amount::new(amount, currency)
+	}
+	fn setup_execute_with_vault(
+		issue_amount: Balance,
+		issue_fee: Balance,
+		griefing_collateral: Balance,
+		amount_transferred: Balance,
+		vault: DefaultVaultId<Test>,
+	) -> Result<H256, DispatchError> {
+		let vault_clone = vault.clone();
+		let vault_clone_2 = vault.clone();
```

Comment:
This is something I did not fully understand but was "required" by the compiler. I was going to look into it to improve this since it's not very elegant.

Reply:
So after exploring a bit more, the first clone is needed since we are going to pass vault to 2 closures, that is expected. The second one inside the closure is because each closure is an FnMut, which means that if it were to take ownership even if it is a clone of vault and is not used any more after that, it would not be available for any further call to the closure. 
Hope it makes sense!

Reply:
Makes sense, thanks for looking into this :+1: